utils/utf8.cc

```diff
@@ -491,16 +494,25 @@ bool validate(const uint8_t *data, size_t len) {
     }
 
     // Check remaining bytes with naive method
-    return validate_naive(data, len);
+    return validate_naive(data, len) == -1;
 }
 
 #else
 // No SIMD implementation for this arch, fallback to naive method
 bool validate(const uint8_t *data, size_t len) {
-    return validate_naive(data, len);
+    return validate_naive(data, len) == -1;
 }
 #endif
 
+ssize_t validate_with_error_position(const uint8_t *data, size_t len) {
```

Comment:
You could use `std::optional<size_t>` as the return type and indicate lack of error with `std::nullopt`.

Reply:
I agree that for the public interface it's better to avoid magic return values.

Reply:
Changed to `std::optional` in amended commit.

Found that previously I forgot to fix one unit test (checking for hardcoded exception message) - it is now fixed and all unit tests pass correctly.

https://github.com/scylladb/scylla/pull/7169/files#diff-43be90fff306ebda1a2855a78a22daa1L577-R577